Content.Server/Physics/Controllers/MoverController.cs

```diff
@@ -95,44 +89,121 @@ private void HandleShuttleMovement(SharedPlayerInputMoverComponent mover)
 
                     speedCap = _shuttleDockSpeedCap;
                     break;
                 case ShuttleMode.Cruise:
-                    if (movement.Length != 0.0f)
+                    if (movement.Y == 0f)
+                    {
+                        system.DisableThrustDirection(shuttleComponent, DirectionFlag.South);
+                        system.DisableThrustDirection(shuttleComponent, DirectionFlag.North);
+                    }
+                    else
                     {
+                        var direction = movement.Y > 0f ? Direction.North : Direction.South;
+                        var linearSpeed = shuttleComponent.LinearThrusterImpulse[(int) direction / 2];
+
                         if (physicsComponent.LinearVelocity.LengthSquared == 0f)
                         {
-                            movement.Y *= 5f;
+                            linearSpeed *= 5f;
                         }
 
                         // Currently this is slow BUT we'd have a separate multiplier for docking and cruising or whatever.
-                        physicsComponent.ApplyLinearImpulse((physicsComponent.Owner.Transform.WorldRotation + new Angle(MathF.PI / 2)).ToVec() *
-                                                            shuttleComponent.SpeedMultipler *
-                                                            physicsComponent.Mass *
-                                                            movement.Y *
-                                                            2.5f);
+                        physicsComponent.ApplyLinearImpulse(physicsComponent.Owner.Transform.WorldRotation.Opposite().ToWorldVec() *
+                                                            linearSpeed *
+                                                            movement.Y);
 
-                        physicsComponent.ApplyAngularImpulse(-movement.X * angularSpeed * physicsComponent.Mass);
+                        switch (direction)
+                        {
+                            case Direction.North:
+                                system.DisableThrustDirection(shuttleComponent, DirectionFlag.South);
+                                system.EnableThrustDirection(shuttleComponent, DirectionFlag.North);
+                                break;
+                            case Direction.South:
+                                system.DisableThrustDirection(shuttleComponent, DirectionFlag.North);
+                                system.EnableThrustDirection(shuttleComponent, DirectionFlag.South);
+                                break;
+                        }
+                    }
+
+                    var angularSpeed = shuttleComponent.AngularThrust;
+
+                    if (movement.X == 0f)
+                    {
+                        system.DisableThrustDirection(shuttleComponent, DirectionFlag.West);
+                        system.DisableThrustDirection(shuttleComponent, DirectionFlag.East);
+                    }
```

Comment:
should maybe check if its sufficiently close to 0f? same with movement.Y check

Reply:
Wouldn't that eat movement inputs in some scenarios?

Reply:
eh, its fine then